invenio_app_ils/ui/backoffice/src/pages/LoanDetails/state/tests/actions.test.js

```diff
@@ -0,0 +1,159 @@
+import configureMockStore from 'redux-mock-store';
+import thunk from 'redux-thunk';
+import MockAdapter from 'axios-mock-adapter';
+import axios from 'axios';
+// Actions to be tested
+import * as actions from '../actions';
+import { initialState } from '../reducer';
+import { serializeLoanDetails } from '../selectors';
+import * as types from '../types';
+
+const middlewares = [thunk];
+const mockStore = configureMockStore(middlewares);
+const loansBaseUrl = 'https://localhost:5000/api/circulation/loans';
+
+describe('loan actions', () => {
+  let store;
+  beforeEach(() => {
+    store = mockStore(initialState);
+    store.clearActions();
+  });
+
+  describe('fetch loan details', () => {
+    let mock;
+    let response = {};
+    beforeEach(() => {
+      mock = new MockAdapter(axios);
+      mock.onGet(`${loansBaseUrl}/1`).reply(() => {
+        return new Promise((resolve, reject) =>
+          setTimeout(() => {
+            resolve([200, JSON.stringify(response)]);
+          }, 1000)
+        );
+      });
+      mock.onGet(`${loansBaseUrl}/2`).reply(() => {
+        return new Promise((resolve, reject) =>
+          setTimeout(() => {
+            reject([500, 'Error']);
+          }, 1000)
+        );
+      });
+    });
+
+    it('fires a fetch loan details action', done => {
+      const expectedActions = [
+        {
+          type: types.SET_LOAN_FETCH_LOADING,
+        },
+      ];
+
+      store.dispatch(actions.fetchLoanDetails(1)).then(() => {
```

Comment:
Here it is a matter of readibility, as you were saying...
Do we consider more readable the current version or the `return` without `done()`? 
```
return store.dispatch(actions.fetchLoanDetails(1)).then(() => {
        let actions = store.getActions();
        expect(actions[0]).toEqual(expectedActions[0]);
```